```
============================================================ test session starts =============================================================
platform darwin -- Python 3.7.5, pytest-5.3.5, py-1.8.1, pluggy-0.13.1
rootdir: /Users/mcovarr/gitrepos/Testathon-2020/pytest-wdl-tests
plugins: subtests-0.3.0, wdl-1.3.0
collected 1 item                                                                                                                             

tests/expr/declarations/test_expressions_declarations.py F                                                                             [100%]

================================================================== FAILURES ==================================================================
________________________________________________________ test_optional_interpolation _________________________________________________________

workflow_runner = <function workflow_runner.<locals>._run_workflow at 0x103852cb0>

    def test_optional_interpolation(workflow_runner):
        """
        Expression evaluation in declarations.
        """
        inputs = { }
        expected = { }
        workflow_runner(
            "test_expressions_declarations.wdl",
            inputs,
            expected,
            workflow_name = "expressions",
            version = "1.0"
        )
    
>       workflow_runner("test_expressions_declarations.wdl", {}, {})

tests/expr/declarations/test_expressions_declarations.py:15: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
testathon2020/lib/python3.7/site-packages/pytest_wdl/fixtures.py:286: in _run_workflow
    _run_test(executors[0])
testathon2020/lib/python3.7/site-packages/pytest_wdl/fixtures.py:280: in _run_test
    **kwargs
testathon2020/lib/python3.7/site-packages/pytest_wdl/executors/cromwell.py:173: in run_workflow
    wdl_path=wdl_path, import_dirs=self._import_dirs, **kwargs
testathon2020/lib/python3.7/site-packages/pytest_wdl/executors/__init__.py:359: in get_target_name
    wdl_doc = parse_wdl(wdl_path, **kwargs)
testathon2020/lib/python3.7/site-packages/pytest_wdl/executors/__init__.py:316: in parse_wdl
    check_quant=check_quant
testathon2020/lib/python3.7/site-packages/WDL/Tree.py:1345: in load
    import_max_depth=import_max_depth,
/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/asyncio/base_events.py:579: in run_until_complete
    return future.result()
testathon2020/lib/python3.7/site-packages/WDL/Tree.py:1288: in load_async
    doc = _parser.parse_document(read_rslt.source_text, uri=uri, abspath=read_rslt.abspath)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

txt = 'version 1.0\n\nstruct Person {\n    String name\n    Int age\n}\n\ntask exprs {\n    String string = "string"\n    St...= booleans\n        Array[Boolean] output_booleans_ = booleans_\n    }\n}\n\nworkflow expressions {\n  call exprs\n}\n'
version = '1.0', uri = '/Users/mcovarr/gitrepos/Testathon-2020/pytest-wdl-tests/tests/expr/declarations/test_expressions_declarations.wdl'
abspath = '/Users/mcovarr/gitrepos/Testathon-2020/pytest-wdl-tests/tests/expr/declarations/test_expressions_declarations.wdl'

    def parse_document(
        txt: str, version: Optional[str] = None, uri: str = "", abspath: str = ""
    ) -> Tree.Document:
        npos = SourcePosition(uri=uri, abspath=abspath, line=0, column=0, end_line=0, end_column=0)
        if not txt.strip():
            return Tree.Document(npos, [], {}, [], None,)
        if version is None:
            # for now assume the version is 1.0 if the first line is "version <number>"
            # otherwise draft-2
            version = "draft-2"
            for line in txt.split("\n"):
                line = line.strip()
                if line and line[0] != "#":
                    if line.startswith("version "):
                        version = line[8:]
                    break
        try:
            (grammar, keywords) = _grammar.get(version)
        except KeyError:
            raise Error.SyntaxError(npos, "unknown WDL version " + version) from None
        try:
            return _DocTransformer(uri=uri, abspath=abspath, keywords=keywords).transform(
                parse(grammar, txt, "document")
            )
        except lark.exceptions.UnexpectedInput as exn:
            pos = SourcePosition(
                uri=(uri if uri else "(buffer)"),
                abspath=(abspath if abspath else "(buffer)"),
                line=getattr(exn, "line", "?"),
                column=getattr(exn, "column", "?"),
                end_line=getattr(exn, "line", "?"),
                end_column=getattr(exn, "column", "?"),
            )
>           raise Error.SyntaxError(pos, str(exn)) from None
E           WDL.Error.SyntaxError: Unexpected token Token(STRING1_FRAGMENT, '- i\n    Int unary = + i\n    Int conditional = if true then 1 else 0\n    Int conditional_ = if true_ then 1 else 0\n    Int mult = i * j\n    Int div = i / j\n    Int mod = i % j\n    Int add = i + j\n    Int sub = i - j\n    Boolean lt = i < j\n    Boolean lte = i <= j\n    Boolean gt = i > j\n    Boolean gte = i >= j\n    Boolean eq = i == j\n    Boolean ne = i != j\n    Boolean and = true_ && false_\n    Boolean or = true_ || false_\n    Person p = { "name": "Jamie", "age": 4 }\n    Array[Boolean] booleans = [ true, false ]\n    Array[Boolean] booleans_ = [ true_, false_ ]\n\n    command {\n      echo noop\n    }\n\n    output {\n        String output_parens = parens\n        String output_left = left\n        Array[String] output_array_string = array_string\n        String output_array_element = array_element\n        Boolean output_negated = negated\n        Boolean output_negated_ = negated_\n        Int output_negated__ = negated__\n        Int output_unary = unary\n        Int output_conditional = conditional\n        Int output_conditional_ = conditional_\n        Int output_mult = mult\n        Int output_div = div\n        Int output_mod = mod\n        Int output_add = add\n        Int output_sub = sub\n        Boolean output_lt = lt\n        Boolean output_lte = lte\n        Boolean output_gt = gt\n        Boolean output_gte = gte\n        Boolean output_eq = eq\n        Boolean output_ne = ne\n        Boolean output_and = and\n        Boolean output_or = or\n        Person output_p = p\n        Array[Boolean] output_booleans = booleans\n        Array[Boolean] output_booleans_ = booleans_\n    }\n}\n\nworkflow expressions {\n  call exprs\n}\n') at line 22, column 21.
E           Expected one of: 
E           	* __ANON_10
E           	* BANG
E           	* LPAR
E           	* FLOAT
E           	* SIGNED_INT
E           	* SIGNED_FLOAT
E           	* LBRACE
E           	* INT
E           	* __ANON_11
E           	* LSQB
E           	* CNAME

testathon2020/lib/python3.7/site-packages/WDL/_parser.py:568: SyntaxError
============================================================= 1 failed in 26.39s =============================================================
```

--- a/pytest-wdl-tests/tests/expr/declarations/test_expressions_declarations.py
+++ b/pytest-wdl-tests/tests/expr/declarations/test_expressions_declarations.py
@@ -1,0 +1,39 @@
+def test_expressions_declarations(workflow_runner):
+    """
+    Expression evaluation in declarations.
+    """
+    inputs = { }
+    expected = {
+        "exprs.output_parens": "string",
+        "exprs.output_left": "left",
+        "exprs.output_array_string": [ "string" ],
+        "exprs.output_array_element": "string",
+        "exprs.output_negated": False,
+        "exprs.output_negated_": False,
+        "exprs.output_negated__": -42,
+        "exprs.output_unary": 42,
+        "exprs.output_conditional": 1,
+        "exprs.output_conditional_": 1,
+        "exprs.output_mult": 1806,
+        "exprs.output_div": 0,
+        "exprs.output_mod": 42,
+        "exprs.output_add": 85,
+        "exprs.output_sub": -1,
+        "exprs.output_lt": True,
+        "exprs.output_lte": True,
+        "exprs.output_gt": False,
+        "exprs.output_gte": False,
+        "exprs.output_eq": False,
+        "exprs.output_ne": True,
+        "exprs.output_and": False,
+        "exprs.output_or": True,
+        "exprs.output_p": {'name': 'Jamie', 'age': 4},
+        "exprs.output_booleans": [ True, False ],
+        "exprs.output_booleans_": [ True, False]
+    }
+    workflow_runner(
+        "test_expressions_declarations.wdl",
+        inputs,
+        expected,
+        workflow_name = "expressions"
+    )

--- a/pytest-wdl-tests/tests/expr/declarations/test_expressions_declarations.wdl
+++ b/pytest-wdl-tests/tests/expr/declarations/test_expressions_declarations.wdl
@@ -1,0 +1,83 @@
+version 1.0
+
+struct Person {
+    String name
+    Int age
+}
+
+task exprs {
+    String string = "string"
+    String parens = (string)
+    String parens_ = ("string")
+    Pair[String, String] pair = ("left", "right")
+    String left = pair.left
+    Array[String] array_string = ["string"]
+    String array_element = array_string[0]
+    Boolean true_ = true
+    Boolean false_ = false
+    Boolean negated = !true_
+    Boolean negated_ = !true
+    Int i = 42
+    Int j = 43
+    Int negated__ = - i
+    Int unary = + i
+    Int conditional = if true then 1 else 0
+    Int conditional_ = if true_ then 1 else 0
+    Int mult = i * j
+    Int div = i / j
+    Int mod = i % j
+    Int add = i + j
+    Int sub = i - j
+    Boolean lt = i < j
+    Boolean lte = i <= j
+    Boolean gt = i > j
+    Boolean gte = i >= j
+    Boolean eq = i == j
+    Boolean ne = i != j
+    Boolean and = true_ && false_
+    Boolean or = true_ || false_
+    Person p = { "name": "Jamie", "age": 4 }
+    Array[Boolean] booleans = [ true, false ]
+    Array[Boolean] booleans_ = [ true_, false_ ]
+
+    command {
+      echo noop
+    }
+
+    runtime {
+      docker: "ubuntu:latest"
+    }
+
+    output {
+        String output_parens = parens
+        String output_left = left
+        Array[String] output_array_string = array_string
+        String output_array_element = array_element
+        Boolean output_negated = negated
+        Boolean output_negated_ = negated_
+        Int output_negated__ = negated__
+        Int output_unary = unary
+        Int output_conditional = conditional
+        Int output_conditional_ = conditional_
+        Int output_mult = mult
+        Int output_div = div
+        Int output_mod = mod
+        Int output_add = add
+        Int output_sub = sub
+        Boolean output_lt = lt
+        Boolean output_lte = lte
+        Boolean output_gt = gt
+        Boolean output_gte = gte
+        Boolean output_eq = eq
+        Boolean output_ne = ne
+        Boolean output_and = and
+        Boolean output_or = or
+        Person output_p = p
+        Array[Boolean] output_booleans = booleans
+        Array[Boolean] output_booleans_ = booleans_
+    }
+}
+
+workflow expressions {
+  call exprs
+}

